Revbump retext: Update PyQt5 deps for >=5.12.

holgersson32644 opened the pull request.

juippis commented on app-editors/retext/retext-7.0.4-r1.ebuild:
```diff
@@ -0,0 +1,73 @@
+# https://bitbucket.org/pypy/compatibility/wiki/Home#!gui-library-bindings
+PYTHON_COMPAT=( python3_{5,6,7} )
+
+inherit distutils-r1 gnome2-utils virtualx xdg-utils
```
gnome2_icon_cache update doesn't work with EAPI-7 anymore. Use xdg_icon_cache_update from xdg-utils instead.

juippis commented on app-editors/retext/retext-7.0.4-r1.ebuild:
```diff
@@ -0,0 +1,73 @@
+"
+DEPEND="
+	${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
With EAPI-7 this is a BDEPEND. distutils-r1 will put it there for you, but doesn't hurt to show it's being used.

holgersson32644 commented:
Hi juippis,

thanks for the review.

> +inherit distutils-r1 gnome2-utils virtualx xdg-utils
> 
> gnome2_icon_cache update doesn't work with EAPI-7 anymore. Use
> xdg_icon_cache_update from xdg-utils instead.

Fixed and finally written down to my checklist for future bumps.

> +DEPEND="
> +	${RDEPEND}
> +	dev-python/setuptools[${PYTHON_USEDEP}]
> 
> With EAPI-7 this is a BDEPEND. distutils-r1 will put it there for you,
> but doesn't hurt to show it's being used.

Fixed.

I also screwed up the removal commit a bit which is fixed now.

mgorny commented on app-editors/retext/retext-7.0.4-r1.ebuild:
```diff
@@ -0,0 +1,75 @@
+	dev-python/python-markdown-math[${PYTHON_USEDEP}]
+	dev-python/PyQt5[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
+	|| (
+		( >=dev-python/PyQt5-5.12 dev-python/PyQtWebEngine )
```
What about PYTHON_USEDEP?

mgorny commented on app-editors/retext/retext-7.0.4-r1.ebuild:
```diff
@@ -0,0 +1,75 @@
+"
+DEPEND="
+	${RDEPEND}
+	test? ( x11-base/xorg-server[xvfb] )
```
Why do you need this explicitly? virtualx should take care of it.

a17r commented on app-editors/retext/retext-7.0.4-r1.ebuild:
```diff
@@ -36,13 +36,18 @@ RDEPEND="
 	dev-python/pygments[${PYTHON_USEDEP}]
 	dev-python/python-markdown-math[${PYTHON_USEDEP}]
-	dev-python/PyQt5[gui,network,printsupport,webengine,widgets,${PYTHON_USEDEP}]
+	dev-python/PyQt5[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
```
This block can be simplified though, right?

* Either more descriptive:
```
|| (
	(
		dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
		>=dev-python/PyQt5-5.12[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
	)
	dev-python/PyQt5[gui,network,printsupport,webengine,widgets,${PYTHON_USEDEP}]
)
```
* Or compact:
```
dev-python/PyQt5[gui,network,printsupport,widgets,${PYTHON_USEDEP}]
|| (
	dev-python/PyQtWebEngine[${PYTHON_USEDEP}]
	>=dev-python/PyQt5-5.12[webengine,${PYTHON_USEDEP}]
)
```

Please also note: One line per dependency.

a17r commented:
ping ;)
